src/parse/string/reduce/shared.ts

```diff
@@ -64,8 +65,10 @@ export const writeMultipleLeftBoundsMessage = <
     `An expression may have at most one left bound (parsed ${openLimit}${invertedComparators[openComparator]}, ${limit}${invertedComparators[comparator]})`
 
 export type writeMultipleLeftBoundsMessage<
-    openLimit extends NumberLiteral,
+    openLimit extends ValidLiteral,
     openComparator extends MinComparator,
-    limit extends NumberLiteral,
+    limit extends ValidLiteral,
     comparator extends MinComparator
 > = `An expression may have at most one left bound (parsed ${openLimit}${InvertedComparators[openComparator]}, ${limit}${InvertedComparators[comparator]})`
+
+export type ValidLiteral = NumberLiteral | DateLiteral
```

Comment:
`ValidLiteral` is being used everywhere and the name sucks KEKW

It should probably be `LimitLiteral`. In general, prefixing `Valid` to a type doesn't actually change its meaning, so it's basically just equivalent to `Literal`.

Also, I notice it is casted very often when it's used. See if you can use some kind of type guard to eliminate those. It's a fairly straightforward type so in most situations, we shouldn't need to cast it e.g. to pass it to an error message.